src/js/controllers/oppController.js

```diff
@@ -62,149 +62,111 @@ export const oppScreen = async function(state){
+                        flag = true;
+                    }
                 }
-            });
-
-            //console.log("attention: " + state.user.uid);
-            //console.log("attention: " + state.opp.opps[btn.id].data().registered[1].uid);
-            let flag = false;
-            for (let i=0; i<state.opp.opps[btn.id].data().registered.length; i++){
-                if (state.opp.opps[btn.id].data().registered[i].uid == state.user.uid){
-                    flag = true;
+                if (flag){
+                    document.querySelector('.ref-request').innerHTML = `
+                    <div class="ref-request" id="1">
+                        <button type="button">Request Sent!</button>
+                    </div>
+                    `;
+                } else {
+                    document.querySelector('.ref-request').innerHTML = `
+                    <div class="ref-request" id="0">
+                        <button type="button">Request a Referral</button>
+                    </div>
+                    `;
                 }
-            }
-            if (flag){
-                document.querySelector('.ref-request').innerHTML = `
-                <div class="ref-request" id="1">
-                    <button type="button">Request Sent!</button>
-                </div>
-                `;
-            } else {
-                document.querySelector('.ref-request').innerHTML = `
-                <div class="ref-request" id="0">
-                    <button type="button">Request a Referral</button>
-                </div>
-                `;
-            }
 
 
-            document.querySelector('.ref-request').addEventListener('click', e => {
-                const btn3 = e.target.closest('.ref-request');
-                console.log("attention: "+state.opp.opps[btn.id].id);
+                document.querySelector('.ref-request').addEventListener('click', e => {
+                    const btn3 = e.target.closest('.ref-request');
+                    console.log("attention: "+state.opp.opps[btn.id].id);
+
+                    if(btn3 && btn3.id == 0) {
+
+                        
+                        // UI change
+                        btn3.innerHTML = `
+                            <div class="ref-request" id="1">
+                                <button type="button">Request Sent!</button>
+                            </div>
+                        `;
+                        let tempArr = state.opp.opps[btn.id].data().registered;
+                        tempArr.push({
+                            name: state.home.profile.name,
+                            uid: state.user.uid
+                        });
+                        //console.log(tempArr);
+                        //console.log(state.opp.opps[btn.id].data().registered);
+                        db.collection('Opportunities').doc(state.opp.opps[btn.id].id).update({
+                            registered: tempArr
+                        });
```

Comment:
If I click on the button multiple times, I am constantly added to the list. I now appear 3 times in the registered array